FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,49 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		perform { context in
+			do {
+				if let cache = try CoreDataCache.find(in: context) {
+					completion(.found(feed: cache.localFeed, timestamp: cache.timestamp!))
```

Comment:
Please remove force unwrapping since it is not allowed by the instructions
```suggestion
					completion(.found(feed: cache.localFeed, timestamp: cache.timestamp))
```